Analysis/PIDTools/include/Compute_dEdxProcessor.hh

```diff
@@ -23,12 +24,20 @@ private:
   //ComputeddEdx *_mydEdx;
   float *CalculateEnergyLoss(TrackerHitVec& hitVec, Track* trk);  
   float getNormalization(double dedx, float hit, double trkcos);
+  float getSmearing(float dEdx);
 
   float _TPC_inner;
   std::string _description;
   std::string _LDCTrackCollection;
   float _energyLossErrorTPC;
   LCCollection* _LDCCol;
+
+  //for smearing
+  std::random_device seed_gen;
+  std::default_random_engine *engine;
+  std::uniform_real_distribution<> dist;
+  bool _isSmearing;
+  float _smearingFactor;
```

Comment:
You are introducing new warnings, please fix them.
```
../Analysis/PIDTools/src/Compute_dEdxProcessor.cc:29:1: warning: 'Compute_dEdxProcessor::seed_gen' should be initialized in the member initialization list [-Weffc++]
../Analysis/PIDTools/src/Compute_dEdxProcessor.cc:29:1: warning: 'Compute_dEdxProcessor::engine' should be initialized in the member initialization list [-Weffc++]
../Analysis/PIDTools/src/Compute_dEdxProcessor.cc:29:1: warning: 'Compute_dEdxProcessor::dist' should be initialized in the member initialization list [-Weffc++]
../Analysis/PIDTools/src/Compute_dEdxProcessor.cc:29:1: warning: 'Compute_dEdxProcessor::_isSmearing' should be initialized in the member initialization list [-Weffc++]
../Analysis/PIDTools/src/Compute_dEdxProcessor.cc:29:1: warning: 'Compute_dEdxProcessor::_smearingFactor' should be initialized in the member initialization list [-Weffc++]
```

If you could look at the rest of the warnings pertaining to `Compute_dEdxProcessor` would be appreciated